discovery/azure/azure.go

```diff
@@ -71,6 +72,9 @@ type Discovery struct {
 
 // NewDiscovery returns a new AzureDiscovery which periodically refreshes its targets.
 func NewDiscovery(cfg *config.AzureSDConfig, logger log.Logger) *Discovery {
+	if logger == nil {
+		logger = log.NewNopLogger()
```

Comment:
I'm not sure logging nowhere if a developer forgets to pass in a logger is the most robust choice. I'd argue it's better to send logs somewhere rather than nowhere if there's a production problem being debugged.